background.ts

```diff
@@ -193,8 +189,23 @@ Electron.ipcMain.on('confirm-do-image-deletion', async(event, imageName, imageID
 
   if (choice === 0) {
     try {
-      await imageManager.deleteImage(imageID);
-      refreshImageList();
+      const maxNumAttempts = 2;
+      // On macOS a second attempt is needed to actually delete the image.
+      // Probably due to a timing issue on the server part of kim, but not determined why.
+      // Leave this in for windows in case it can happen there too.
+      let i = 0;
+
+      for (i = 0; i < maxNumAttempts; i++) {
+        await imageManager.deleteImage(imageID);
+        await imageManager.refreshImages();
+        if (!imageManager.listImages().find(image => image.imageID === imageID)) {
```

Comment:
```suggestion
        if (!imageManager.listImages().some(image => image.imageID === imageID)) {
```
It's _slightly_ more clear that we're looking for a match, and not actually caring about the image. Not by much, but still.

Reply:
ok